### DISCUSSION TITLE
Issue with Session not saving values anymore

### DISCUSSION THREAD

**Discussion by pneves001** [laravel/framework#40200] [Q&A] [answered]
- Laravel Version:  "^8.12",
- PHP Version:8.1.1
- Database Driver & Version: pgsql   14.1. 
- Session Store: redis 

### Description:
I just did a composer update on my laravel install a few days ago. Ever since the Session session hasn't been working. It's manifested itself in several ways.
  
First the  when attempting to do a return redirect()->back()->withErrors($validator->errors()); The following code doesn't display any errors. 
```

@if ($errors->any())
    <div class="alert alert-danger">
        <ul>
            @foreach ($errors->all() as $error)
                <li>{{ $error }}</li>
            @endforeach
        </ul>
    </div>
@endif
```

This used to work before the update. And If I have to go and now change it to some other way or using some other format it will be a monumental undertaking on my project. We are talking about megabytes of source code here that I do not have to time to rewrite. 

Other issues I'm running into are that the Session::put() function is not saving arrays of data. And I've also tried adding  Session::save() call and still I'm getting no results. 

I'm also having issues with the laravel authentication system as well. I've tried to use the database session table as a cache rather than redis and it wasn't writing to the database.  I've tried setting it so it saves the session in redis. No dice.  

### Steps To Reproduce:

I'm not sure how this happened. It just stopped working all of a sudden and nothing I've tried seems to fix it.

**Comment by GertjanRoke** [comment, marked as answer]
@pneves001 Could you check if this middleware is still used on the routes where it goes wrong?
```
\Illuminate\View\Middleware\ShareErrorsFromSession::class
```

**Reply by pneves001** [reply]
It was commented out.  I had to turn it back on.  But it had no effect.

**Reply by GertjanRoke** [reply]
> It was commented out.  I had to turn it back on.  But it had no effect. 

You probably did, but after enabling that middleware you also ran `php artisan route:clear`?

**Reply by pneves001** [reply]
I tried running php artisan route:clear. It seems to have fixed the session stuff.  However, I'm still having an issue with the authentication. It seems to be redirecting back to my login screen instead of logging in to my dashboard screen. I just want to point to point out I haven't made any changes to this code prior to all this happening. Its just not redirecting to the right place.  

To authenticate I'm using the following logic. 

```
if (Auth::attempt(['email' => trim(strtolower($email)), 'password' => trim($password)])) 
{
return redirect($this->redirectPath);
}	

```
It gets to the redirect and just goes back to the login screen.

**Reply by GertjanRoke** [reply]
Did you update from Laravel 7 to 8 btw? Because they made some changes to where Laravel redirects after a successful login.

You could always add a "after" middleware to get the stack trace to to give you some more info.

I don't know if you're on Discord else I can maybe look with you to debug this problem.
My Discord account is: Gertjan#1639

**Reply by pneves001** [reply]
I'm not on this discord channel. But I can get on to it and then hit you up.

**Reply by pneves001** [reply]
@GertjanRoke,  I joined the laravel discord group. But I am unable to find your account on there.  However, what you are saying just gave me an idea about what is going on. It looks like it might be the update messed up my email verification  route.

**Reply by GertjanRoke** [reply]
Could you say hi in general, then I can find you

**Reply by pneves001** [reply]
Ok this is solved. It turns out there were some settings I set in the config files that was causing all my issues where the domains didn't match and was causing the pages not to load properly.

**Reply by LogicSatinn** [reply]
are you using Multi-domain Tenancy?? I'm facing the exact problem. I cannot figure out what keeps going wrong. Perhaps you can share what you fixed so i can try fixing from my end.

**Reply by pneves001** [reply]
Oh this was 3 months ago. I don't remember what I did to fix this now. The domain in your session needs to match your website domain though.
